include/bluetooth/mesh/main.h

```diff
@@ -272,6 +286,23 @@ int bt_mesh_input_string(const char *str);
  */
 int bt_mesh_input_number(uint32_t num);
 
+/** @brief Register a callback to retrieve a local public key.
+ *
+ * This function needs to be used when @ref BT_MESH_PUB_KEY_OOB is set.
+ * The retrieved public key has to be shared with a Provisioner using
+ * any available OOB technology.
+ *
+ * A new key is generated after each provisioning attempt.
+ *
+ * Used only by Provisionee.
+ *
+ * @param cb Callback that will be called when the public key is ready.
+ * Returns a local 64-bytes long public key generated for the provisioning.
+ *
+ * @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_prov_local_pub_key_cb_register(void (*cb)(const uint8_t *key));
```

Comment:
Why implement such a function?

This function only provides public key capabilities, and does not have private key information,
so even if the App has a public key, it's still cannot use it for `prov->private_key`. 

https://github.com/zephyrproject-rtos/zephyr/blob/fa57145f180e1249948b273c7e2dc006ce53f21e/subsys/bluetooth/mesh/prov_device.c#L320

So public/private key can only be static.

Reply:
My suggestion is that the App layer defines its own way of generating public and private keys. The protocol stack does not provide capabilities. Therefore, this registration can be omitted.